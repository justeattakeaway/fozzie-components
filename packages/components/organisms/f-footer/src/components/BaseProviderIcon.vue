<template>
    <component :is="iconType" />
</template>

<script>
// Payment icons
import {
    PaymentAmexIcon as AmexIcon,
    PaymentAmexSafekeyIcon as AmexSafekeyIcon,
    PaymentDankortIcon as DankortIcon,
    PaymentInteracIcon as InteracIcon,
    PaymentMaestroIcon as MaestroIcon,
    PaymentMastercardIcon as MastercardIcon,
    PaymentMastercardSecurecodeIcon as MastercardSecurecodeIcon,
    PaymentPaypalIcon as PaypalIcon,
    PaymentVisaIcon as VisaIcon,
    PaymentVisaTransparentNarrowIcon as VisaVerifiedIcon
} from '@justeat/f-vue-icons';

// Social icons
import {
    SocialFacebookIcon as FacebookIcon,
    SocialTwitterIcon as TwitterIcon,
    SocialYoutubeIcon as YoutubeIcon,
    SocialPinterestIcon as PinterestIcon,
    SocialInstagramIcon as InstagramIcon,
    RssIcon
} from '@justeattakeaway/pie-icons-vue';

import iconPropsMixin from '../mixins/iconProps.mixin';

export default {
    components: {
        FacebookIcon,
        TwitterIcon,
        YoutubeIcon,
        PinterestIcon,
        InstagramIcon,
        RssIcon,
        AmexIcon,
        AmexSafekeyIcon,
        DankortIcon,
        InteracIcon,
        MaestroIcon,
        MastercardIcon,
        MastercardSecurecodeIcon,
        PaypalIcon,
        VisaIcon,
        VisaVerifiedIcon
    },

    mixins: [
        iconPropsMixin
    ],

    computed: {
        iconType () {
            return `${this.name}-icon`;
        }
    }
};
</script>
