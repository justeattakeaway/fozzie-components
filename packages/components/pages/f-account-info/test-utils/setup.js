import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
