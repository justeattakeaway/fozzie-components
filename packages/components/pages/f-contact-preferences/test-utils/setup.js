export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
