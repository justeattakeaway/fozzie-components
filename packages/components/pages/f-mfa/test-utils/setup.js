export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
