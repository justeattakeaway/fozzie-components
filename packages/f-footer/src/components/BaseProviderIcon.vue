<template>
    <component :is="iconType" />
</template>

<script>
// Social and Payment icons
import {
    SocialFacebookIcon as FacebookIcon,
    SocialTwitterIcon as TwitterIcon,
    SocialYoutubeIcon as YoutubeIcon,
    SocialPinterestIcon as PinterestIcon,
    SocialInstagramIcon as InstagramIcon,
    SocialRssIcon as RssIcon,
    PaymentAmexIcon as AmexIcon,
    PaymentAmexSafekeyIcon as AmexSafekeyIcon,
    PaymentDankortIcon as DankortIcon,
    PaymentInteracIcon as InteracIcon,
    PaymentMaestroIcon as MaestroIcon,
    PaymentMastercardIcon as MastercardIcon,
    PaymentMastercardSecurecodeIcon as MastercardSecurecodeIcon,
    PaymentPaypalIcon as PaypalIcon,
    PaymentVisaIcon as VisaIcon,
    PaymentVisaVerifiedIcon as VisaVerifiedIcon
} from '@justeat/f-vue-icons';

export default {
    components: {
        FacebookIcon,
        TwitterIcon,
        YoutubeIcon,
        PinterestIcon,
        InstagramIcon,
        RssIcon,
        AmexIcon,
        AmexSafekeyIcon,
        DankortIcon,
        InteracIcon,
        MaestroIcon,
        MastercardIcon,
        MastercardSecurecodeIcon,
        PaypalIcon,
        VisaIcon,
        VisaVerifiedIcon
    },

    props: {
        name: {
            type: String,
            required: true
        }
    },

    computed: {
        iconType () {
            return `${this.name}-icon`;
        }
    }
};
</script>
