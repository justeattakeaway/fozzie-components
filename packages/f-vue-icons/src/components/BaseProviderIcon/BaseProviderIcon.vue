<template>
    <component
        :is="iconType" />
</template>

<script>
// Social
import FacebookIcon from '../social/FacebookIcon/FacebookIcon.vue';
import TwitterIcon from '../social/TwitterIcon/TwitterIcon.vue';
import YoutubeIcon from '../social/YoutubeIcon/YoutubeIcon.vue';
import PinterestIcon from '../social/PinterestIcon/PinterestIcon.vue';
import InstagramIcon from '../social/InstagramIcon/InstagramIcon.vue';
import RssIcon from '../social/RssIcon/RssIcon.vue';
// Payments
import AmexIcon from '../payment/AmexIcon/AmexIcon.vue';
import AmexSafekeyIcon from '../payment/AmexSafekeyIcon/AmexSafekeyIcon.vue';
import DankortIcon from '../payment/DankortIcon/DankortIcon.vue';
import InteracIcon from '../payment/InteracIcon/InteracIcon.vue';
import MaestroIcon from '../payment/MaestroIcon/MaestroIcon.vue';
import MastercardIcon from '../payment/MastercardIcon/MastercardIcon.vue';
import MastercardSecurecodeIcon from '../payment/MastercardSecurecodeIcon/MastercardSecurecodeIcon.vue';
import PaypalIcon from '../payment/PaypalIcon/PaypalIcon.vue';
import VisaIcon from '../payment/VisaIcon/VisaIcon.vue';
import VisaVerifiedIcon from '../payment/VisaVerifiedIcon/VisaVerifiedIcon.vue';

export default {
    components: {
        FacebookIcon,
        TwitterIcon,
        YoutubeIcon,
        PinterestIcon,
        InstagramIcon,
        RssIcon,
        AmexIcon,
        AmexSafekeyIcon,
        DankortIcon,
        InteracIcon,
        MaestroIcon,
        MastercardIcon,
        MastercardSecurecodeIcon,
        PaypalIcon,
        VisaIcon,
        VisaVerifiedIcon
    },
    props: {
        name: {
            type: String,
            required: true
        }
    },
    computed: {
        iconType () {
            return `${this.name}-icon`;
        }
    }
};
</script>
