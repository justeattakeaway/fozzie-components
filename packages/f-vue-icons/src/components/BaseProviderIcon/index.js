import BaseProviderIcon from './BaseProviderIcon.vue';

const install = Vue => {
    Vue.component(BaseProviderIcon.name, BaseProviderIcon);
};

export default install;
