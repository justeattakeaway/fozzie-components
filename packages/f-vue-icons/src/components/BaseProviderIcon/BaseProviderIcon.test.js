import { shallowMount } from '@vue/test-utils';
import BaseProviderIcon from './BaseProviderIcon.vue';

let propsData = { name: 'facebook' };

describe('BaseProviderIcon', () => {
    it('should be defined', () => {
        const wrapper = shallowMount(BaseProviderIcon, {
            propsData
        });
        expect(wrapper).toBeDefined();
    });

    it('default component markup is rendered', () => {
        // Arrange & Act
        const wrapper = shallowMount(BaseProviderIcon, {
            propsData
        });

        // Assert
        expect(wrapper).toMatchSnapshot();
    });

    it('Maestro card component should be rendered if "maestro" name is passed as a prop', () => {
        // Arrange
        propsData = { name: 'maestro' };
        const wrapper = shallowMount(BaseProviderIcon, {
            propsData
        });


        // Act
        const result = wrapper.vm.iconType;

        // Assert
        expect(result).toBe('maestro-icon');
    });
});
