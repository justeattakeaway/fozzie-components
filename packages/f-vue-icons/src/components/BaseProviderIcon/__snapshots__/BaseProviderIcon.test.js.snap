// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseProviderIcon default component markup is rendered 1`] = `<facebook-icon-stub></facebook-icon-stub>`;
